src/overlords/offense/roomPoisoner.ts

```diff
@@ -0,0 +1,159 @@
+			return;
+		}
+		// Ensure you are in the assigned room
+		if (roomPoisoner.room == this.room && !roomPoisoner.pos.isEdge) {
+			//corner case: unclaimed controller blocked, while sources not 100% blocked
+			if(!this.room.my && this.sourcesWallSites && this.controllerWallSites &&
+				this.controllerWallSites.length == 0 &&  this.sourcesWallSites.length > 0){
+				
+				const dismantleTarget = this.findStructureBlockingController(roomPoisoner);
+				if (dismantleTarget) {
+					roomPoisoner.task = Tasks.dismantle(dismantleTarget);
+					return;
+				}	
+			}
+
+			// upgrade controller to level 2 to unlock walls			
+			if (this.room && this.room.controller && this.room.my &&
+					   (this.room.controller.level < 2) &&
+					   !(this.room.controller.upgradeBlocked > 0)) {
+				roomPoisoner.task = Tasks.upgrade(this.room.controller);
+				return;
+			} 
+			//repair poison walls < 1000 hits, assuming all other wall are destoryed by directive
+			let wallsToRepair = _.filter(this.room.walls, wall => wall.hits < 1000);
+			if(wallsToRepair.length){
+				let closestWall = roomPoisoner.pos.findClosestByRange(wallsToRepair);
+				if(closestWall){
+					roomPoisoner.task = Tasks.repair(closestWall);
+					return;
+				}
+			}
+			//build wall csites for controller then sources.
+			if (this.controllerWallSites && this.controllerWallSites.length) {
+				roomPoisoner.task = Tasks.build(this.controllerWallSites[0]);
+			} else if (this.sourcesWallSites && this.sourcesWallSites.length) {
+				roomPoisoner.task = Tasks.build(this.sourcesWallSites[0]);
+			} 
+		} else {
+			roomPoisoner.goTo(this.pos, {ensurePath: true, avoidSK: true});
+		}
+	}
+
+	private handleAntiController(antiController: Zerg): void {					
+		if (antiController.room == this.room && !antiController.pos.isEdge) {
+			if(!antiController.pos.isNearTo(this.room.controller!)){
+                antiController.goTo(this.room.controller!);
+            }
+			//kill claimer if room claimed, it can be blocking wall csite creation
+			if (this.room.my && this.room!.controller!.level == 2){
+				antiController.suicide();
+				return;
+			}
+			//counter reserver if reserved by enemy. else, claim it
+			if (this.room && this.room.controller && this.room.controller.reservation && this.room.controller.reservation.username != MY_USERNAME) {
+				antiController.attackController(this.room.controller);
+			} else if (this.room && !this.room.my){
+				antiController.task = Tasks.claim(this.room.controller!);
+			}
+		} else {
+			// reserver.task = Tasks.goTo(this.pos);
+			antiController.goTo(this.pos);
+		}
+
+		//suspend immediatly to avoid colonization flag, (note: tried it in init() of directive but it missed it by one tick)
+		if(antiController.room == this.room && this.room.my){
+			const colonyMemory = Memory.colonies[this.room.name] as ColonyMemory | undefined;
+			if (colonyMemory && !colonyMemory.suspend) {
+				OvermindConsole.suspendColony(this.room.name);
```

Comment:
I'll add a `Colony.suspend()` method soon so you don't have to access `OvermindConsole`